Fixed commit order

tomaszgaw opened the pull request:
Fixes visit method of CommitOrderCalculator.

Without the fix the insertion order is not properly calculated when a cycle within the graph involves more than 2 classes, leading to a violation of not null constraints on the owning side of ManyToOne relation. 

With the fix when a cycle is detected a node is revisited ignoring edges of weight 0.

SenseException committed:
Set copy-method as deprecated

SenseException committed:
Add deprecation of EntityManagerInterface::copy() to upgrade information

Majkl578 committed:
Added deprecation warnings for 2.x

Majkl578 committed:
Upgrading notes for 2.7

Majkl578 committed:
Bump version to 2.7-dev

SenseException committed:
Create 2.7 upgrade headline for deprecation changes

someniatko committed:
Split and deprecate AbstractQuery#useResultCache()

holtkamp committed:
Mention that lifecycle callbacks do not support Embeddables

As discussed in https://github.com/doctrine/doctrine2/issues/6855

Majkl578 committed:
Update homepage

Majkl578 committed:
Merge pull request #7382 from Majkl578/homepage-2.7

Update homepage

m-r-r committed:
Add a failing test for issue #7505

Majkl578 committed:
CI: Test against PHP 7.4snapshot instead of nightly (8.0)

Ocramius committed:
Merge pull request #7600 from Majkl578/travis-php7.4-2.7

[2.7] CI: Test against PHP 7.4snapshot instead of nightly (8.0)

tomaszgaw committed:
Fix CommitOrderCalculator::visit()

This fixes a commit order when persisting many entities with cyclic
associations within one flush. Before the fix, when multiple cycles
were presented in the precedence graph and more than two classes
existed in within a cycle the calculated commit order was likely to be
wrong, resulting in not null constraint violation when saving to
a database. An example of such situation is provided in
CommitOrderCalculatorTest::testCommitOrdering4() - this is a test case
with won’t pass without the introduced fix.

Prior to the fix the weight of an edge in a class precedence graph was
compared only to the weight of the subsequent edges. Therefore, in case
of cycles containing more than two classes the result was
unpredictable. The fixed code uses also the weight of further nodes in
the precedence graph. It is achieved by adding minimum weight argument
to the visitor function of CommitOrderCalulator – edges having weight
below the minimum weight are ignored. The minimum weight is increased
each time a node is revisited, i. e. a cycle is detected. It guarantees
that edges of highest priority representing not nullable relationships
are processed first.

SenseException commented:
Can you please cover your changes with tests?

I'm also not sure yet if this introduces changes in the behaviour on first look.

tomaszgaw commented:
The test I added passes when runs against the fixed code and fails when runs against the original code.

greg0ire commented on tests/Doctrine/Tests/ORM/CommitOrderCalculatorTest.php:
```diff
@@ -100,6 +100,33 @@ public function testCommitOrdering3()
         $this->assertContains($sorted, $correctOrders, '', false, true, true);
     }
+    
+    public function testCommitOrdering4()
```
I can't say I really understand this PR, but here is some help for those who can:

```
       +---------+
       |         |
    +--+ Class 4 <---+
    |  |         |   |
  1 |  +---------+   | 0
    |                |
    v                |
+---+-----+    +-----+---+
|         |    |         |
| Class 1 +--->+ Class 2 |
|         | 0  |         |
+---+-----+    +-----+---+
    ^                |
    |                |
  0 |                | 0
    |  +---------+   |
    |  |         |   |
    +--+ Class 3 +<--+
       |         |
       +---------+

```

Maybe embed this as a comment if you find it useful?

tomaszgaw replied:
Thank you for the suggestion. I embedded this graph in a comment within the test.

greg0ire commented:
There should be only one commit, the test does not make sense without the code, and the code should not come without tests. Please improve the commit message to make as interesting as the pull request text.

greg0ire commented on lib/Doctrine/ORM/Internal/CommitOrderCalculator.php:
```diff
@@ -149,11 +149,15 @@ public function sort()
      * @param \stdClass $vertex
      */
-    private function visit($vertex)
+    private function visit($vertex, $minimumWeight = 0)
```
This branch requires 7.2, you can use a type hint:

```suggestion
    private function visit($vertex, int $minimumWeight = 0)
```

tomaszgaw replied:
I applied the suggested change in the last (squashed) commit. Thank you.

rtek committed:
Prettify arrays in orm:mapping:describe command

This will prevent excessive column width and wrapping in the output which uses Symfony\Component\Console\Style\SymfonyStyle::table().

someniatko committed:
Make ResultCacheTest tests slightly more logical

nicolas-grekas committed:
Allow Symfony 5.0

jwage committed:
Merge pull request #7723 from nicolas-grekas/sf5

Allow Symfony 5.0

Seb33300 committed:
Skip limit subquery if not required

nicolas-grekas committed:
Whitelist existing assets we know about from metadata in SchemaTool::getUpdateSchemaSql()

l-vo committed:
Test asset whitelisting on SchemaTool#getUpdateSchemasSql()

nicolas-grekas committed:
Fix compat of commands with Symfony 5

lcobucci committed:
Merge pull request #7876 from nicolas-grekas/sf5-cmd

Fix compat of commands with Symfony 5

mairo744 committed:
Fix creation of join table names with schemas in SQLite

Join table name doesnt depending on the platform.
Table name was "schema.table" instead of "schema__table".

(cherry picked from commit 4878cd3f4ef30ffc6047c18e0f7b16aafeabc3b4)

Commit:
Fix boolean properties default value when generating entities.

lcobucci committed:
Fix PHP warnings in test suite

lcobucci committed:
Add basic tool to verify deprecation messages

Ocramius committed:
Merge pull request #7901 from lcobucci/add-deprecation-notices

Add deprecation warnings for 2.7.x

Commit:
Fix to pass code quality check.

Commit:
Add test code.

lcobucci committed:
Require more updated doctrine packages

greg0ire committed:
Merge remote-tracking branch 'origin/2.7' into 7900--2.6.x-merge-up-into-2.7

lcobucci committed:
Merge pull request #7900 from doctrine/2.6.x-merge-up-into-2.7

Merge up 2.6 to 2.7

lcobucci committed:
Merge pull request #7863 from Seb33300/skip-limit-subquery

Paginator: Skip limit subquery if not required

lcobucci committed:
Fix deprecation messages version

lcobucci committed:
Add minor BC-break notes on output walkers in paginator

As explained in
https://github.com/doctrine/orm/pull/7863#issuecomment-554578313.

lcobucci committed:
Merge pull request #7701 from someniatko/deprecate-use-result-cache

Split and deprecate AbstractQuery#useResultCache()

lcobucci committed:
Merge pull request #7710 from rtek/pretty-tool-describe

Prettified arrays in tool command orm:mapping:describe

greg0ire committed:
Make sure composer files are valid

The composer.lock is put under version control and it often happens to
be out of sync with the composer.json, which could lead to
hard-to-understand issues.
Using the --strict option here because we might as well aim for a
perfectly valid composer.json

greg0ire committed:
Synchronize lock file and json manifest

lcobucci committed:
Merge pull request #7904 from greg0ire/validate-composer-json

Make sure composer files are valid

lcobucci committed:
Merge remote-tracking branch 'upstream/2.6' into 2.7

lcobucci committed:
Merge remote-tracking branch 'upstream/2.6' into 2.7

lcobucci committed:
Merge pull request #7079 from mairo744/hotfix/sqlite-join-table-name

fix getJoinTableName for sqlite with schema attribute

lcobucci committed:
Add deprecation message for EM#clear($entityName)

lcobucci committed:
Deprecated the usage of number unaware underscore naming strategy

lcobucci committed:
Merge pull request #7909 from lcobucci/add-deprecation-messages

Add deprecation messages

lcobucci committed:
Be explicit about which Doctrine package in message

Avoiding possible confusion while reading the deprecation messages.

lcobucci committed:
Merge pull request #7911 from lcobucci/be-more-explicit-on-deprecation-messages

Be explicit about which Doctrine package in message

lcobucci committed:
Bump up version

lcobucci committed:
Bump up version

Commit:
Fix to pass code quality check.

Commit:
Modified test to use assertTrue.

Commit:
Fix to pass code quality check.

alcaeus committed:
Add upgrade note about schema_filter change

BenMorel committed:
Fix Trying to access array offset on value of type null

beberlei committed:
Revert "#7841 SchemaTool generates extra diff for platforms without FK support"

This reverts commit 3707c3912428423347473a9f20c86a7d858380c4.

beberlei committed:
[GH-7864] Bugfix in PersistentCollection::removeElement for EXTRA_LAZY.

beberlei committed:
[GH-7864] Remove tests that checked invalid behavior.

beberlei committed:
[GH-7864] ExtraLazyCollectionTest is not cacahble and should not fail SLC suite.

beberlei committed:
Housekeeping: phpcbf to fix issues.

beberlei committed:
[GH-7864] Address review comments.

beberlei committed:
Add warning about removeElement on extra lazy

beberlei committed:
[GH-7864] Revert removeElement EXTRA_LAZY support.

lcobucci committed:
Merge pull request #7934 from BenMorel/php74

Fix Trying to access array offset on value of type null

lcobucci committed:
Merge pull request #7937 from doctrine/GH-7930-SqliteForeignKeys

Revert SchemaTool change to check for foreign key support

lcobucci committed:
Fix CS errors

lcobucci committed:
Ignore Doctrine\Common\Persistence\ObjectManagerDecorator deprecation

Since applying the fixes requires bumping up the dependency, which isn't
done in a patch release.

This should be removed in v2.8.0.

lcobucci committed:
Merge pull request #7956 from lcobucci/fix-test-suite

Ignore Doctrine\Common\Persistence\ObjectManagerDecorator deprecation

lcobucci committed:
Merge pull request #7875 from nicolas-grekas/schema-tool

Whilelist existing assets we know about from metadata in SchemaTool::getUpdateSchemaSql()

lcobucci committed:
Remove build.xml and related files

We aren't using ant/phing to handle the releases any more.

lcobucci committed:
Rely on ocramius/package-versions to render the version

Since `Doctrine\ORM\Version` is now deprecated it shall not be updated
on future releases.

This ensures that our CLI tool will present the correct version number.

lcobucci committed:
Merge pull request #7957 from lcobucci/fix-version-information

Fix version information

beberlei committed:
Merge pull request #7894 from TomckySan/convert-default-value-to-boolean

Fix boolean properties default value when generating entities.

DocFX committed:
Just a micro grammar update.

shadowhand committed:
[Docs] Prefer PhpFileCache for caching and remove APC/XCache.

beberlei committed:
Adjust tests back for 2.x.

lcobucci committed:
Merge pull request #7973 from DocFX/patch-1

Just a micro grammar update.

beberlei committed:
Add @group

beberlei committed:
[GH-7505] Fix cs

beberlei committed:
Merge pull request #7974 from beberlei/gh-7505

[GH-7505] Bug in SimpleObjectHydrator when using inheritance with same field

beberlei committed:
Allow everything from ocramius/package-versions ^1.0.

beberlei committed:
[GH-7982] Default Query state to dirty to fix execution of empty query.

beberlei committed:
Downgrade ocramius/package-versions to lowest in composer.lock to support all PHP versions.

greg0ire committed:
Try running phpstan on php 7.4

It might be easier to find packages compatible with both our locked deps
and phpstan with that version of php.

beberlei committed:
Make ocramius/package-versions 1.2 the lowest version for phpstan

beberlei committed:
Merge pull request #7991 from greg0ire/7.4-sa

Try running phpstan on php 7.4

malarzm committed:
Make Embeddable not transient

beberlei committed:
[GH-7982] Bugfix: Passing string|null DQL to Lexer(string $input) leads to downstream notice.

beberlei committed:
Merge pull request #7987 from beberlei/GH-7982-NoSqlExecutor

[GH-7982] no sql executor leads to parse error

beberlei committed:
Merge pull request #7940 from doctrine/GH-7864-ExtraLazyRemoveElement

[GH-7864] Bugfix in PersistentCollection::removeElement for EXTRA_LAZY.

beberlei committed:
Merge pull request #8006 from doctrine/malarzm-patch-1

Make Embeddable not transient

beberlei committed:
Remove nightly builds from .travis.yml

beberlei committed:
Travis: Use 7.4 instead of 7.4snapshot

peterkeatingie committed:
Put into cache using root entity name

beberlei committed:
Merge pull request #8023 from peterkeatingie/query-cache-fix

Put into cache using root entity name

greg0ire committed:
Merge pull request #7230 from holtkamp/patch-2

Mention that lifecycle callbacks do not support Embeddables

RosemaryOrchard committed:
Annotations override naming strategy (#8041)

Add a note/warning that annotations override the naming strategy.

localheinz committed:
Fix: Use neutral pronouns (#8059)

jorrit committed:
Fix documentation of default generated value behavior (#8068)

malarzm committed:
Fix inherited embeddables and nesting after AnnotationDriver change #8006 (#8036)

* Add test case

* Treat parent embeddables as mapped superclasses

* [GH-8031] Bugfix: Get working again on nested embeddables in inherited embeddables.

* Housekeeping: CS

* Update note on limitations

* [GH-8031] Verify assocations still do not work with Embeddables.

* Housekeeping: CS

Co-authored-by: Benjamin Eberlei <kontakt@beberlei.de>

beberlei committed:
[GH-7633] disallow cache partial objects (#8050)

* [GH-7633] Bugfix: Partial queries were stored in 2LC.

There was a check in DefaultQueryCache that prevented partial queries,
because they are not supported. However the checked hint
Query::HINT_FORCE_PARTIAL_LOAD is optional, so cant be used to prevent
caching partial DQL queries.

Introduce a new hint that the SqlWalker sets on detecing a PARTIAL
query and throw an exception in the DefaultQueryCache if thats found.

* Housekeeping: CS

* [GH-7633] HINT_FORCE_PARTIAL_LOAD still needs to be checked.

* Housekeeping: Fix CS

mpdude committed:
Fix regression in 2.7.1 when `mysqli` is used with discriminator column that is not a string (#8055)

* Add a test case showing the regression

* Cast the discriminator value to string

* Fix CS

VincentLanglet committed:
Fix phpdoc (#8074)

arneee committed:
[Docs] Make clear that calling remove() detaches the object (#8081)

I changed a relationship from eager to lazy loading which broker the behaviour of my application in regards to object removal. It was not clear for me that removing an object detaches it and subsequent calls like contains() in a OneToMany relationship with the object scheduled for removal will return false afterwards.

VincentLanglet committed:
Remove TODO (#8078)

MatTheCat committed:
Fix JoinColumn documentation (#7966)

jeroenvdheuvel committed:
Convert PHP to SQL for new object expression (#8062)

maks-rafalko committed:
Warn users about performance issues with DateTime objects in setParameter() (#8114)

binarygondola committed:
removed repository class from metadata (#8125)

The annotation is required only in the next chapter of the tutorial, specifically the "Entity Repositories"

greg0ire committed:
Address deprecations from persistence (#7953)

A backwards-compatibility layer has been added to persistence to help
consumers move to the new namespacing. It is based on class aliases,
which means the type declaration changes should not be a BC-break: types
are the same.
See https://github.com/doctrine/persistence/pull/71

This means:
- using the new namespaces
- adding autoload calls for new types to types that may be extended and
use persistence types in type declarations of non-constructor methods,
so that signature compatibility is recognized by old versions of php.
More details on this at
https://dev.to/greg0ire/how-to-deprecate-a-type-in-php-48cf

greg0ire committed:
Static analysis with Psalm (#8116)

* Remove useless ternaries

If these expressions are truish inside the condition, they will still be
truish inside the if.

* Describe properties more accurately

These are not objects, they are strings holding class names for classes
that implement TreeWalker.

* Remove duplicate key

Comparison::IS and Comparison::EQ are the same. I chose to remove IS
because it does not seem to exist anymore on master

* Remove unwanted . before = operator

This worked, but makes no sense.

* Setup static analysis with Psalm

* Move PHPStan to Github actions

beberlei committed:
Revert changes to embeddable mapping in 2.7 (#8138)

* Revert "Fix inherited embeddables and nesting after AnnotationDriver change #8006 (#8036)"

This reverts commit a9b6b720175fae69e90247e6b7a0e34d7da2f1cc.

* Revert "Make Embeddable not transient"

This reverts commit 58677c29b4bc97273da950f6312e43ff540a7407.

* Housekeeping: CS fixes

greg0ire committed:
Migrate git-phpcs to Github actions (#8146)

orklah committed:
Add psalm types (#7989)

robertbasic committed:
Fix inline code example (#8153)

* Fix inline code example

* Remove extra backslash

greg0ire committed:
Allow doctrine/common 3 and doctrine/persistence 2 (#8158)

albe committed:
TASK: Replace "Blacklist" example with "Banlist" (#8174)

greg0ire committed:
Run benchmark on PHP 7.4 (#8186)

phpbench recently dropped compatibilty with PHP 7.1
When this job was introduced, 7.1 was the latest version of PHP we
supported, so it makes sense to bump to 7.4 now.
See https://github.com/phpbench/phpbench/releases/tag/0.17.0
See e07c90df4488b7b44494695e46c3505f5cf21266

greg0ire committed:
Use CodeCov instead of Scrutinizer  (#8187)

* Document current branch status in the README

* Use CodeCov instead of Scrutinizer

nicolas-grekas committed:
Allow using on PHP 7.1 with Composer 2 (#8184)

vladysta committed:
add dev files to .gitattribute (#8190)

those files probably are not used by end user of the lib

takman1 committed:
Remove unwanted period (#8198)

gquemener committed:
Prohibits class typo in the discriminator map (#8122)

* Prevents incorrect table aliases to be generated

When a defined subclass has a case typo, the query builder will be lost
and will generate exotic table alias. This commit fixes the issue at the
root by prohibiting case typo in discriminator map.

See https://github.com/doctrine/orm/pull/8112 for the consequence of
such typo.

* Controls growing rate of the abstract test class

* Fixes incorrect test case

The Cube class must be autoloaded with the correct case, otherwise
composer autoloader will complain.

* Removes non architecture compliant code

See https://github.com/doctrine/orm/pull/8122/files#r423952247

* Ensures discriminator map is case sensitive

beberlei committed:
[GH-8122] Move test from AbstractMappingDriverTest to ClassMetadataFactoryTest case

greg0ire committed:
Infer return type information with Psalter (#8150)

These changes are a subset of changes done with vendor/bin/psalm --alter
--issues=LessSpecificReturnType
--allow-backwards-incompatible-changes=false

Pictor13 committed:
Add alert to "avoid persisting detached entities" (#8109)

The alert is hidden into the code of ``EntityManager``,
while it's useful to be mentioned in the documentation.

tom93 committed:
Fix QueryBuilder::getParameter() on parameter names with colons (#8107)

* Fix type errors

(partially cherry picked from commit 17e7c2a42e0ee49781e60d6facdd2efc9e1f86b3)

* Fix QueryBuilder::getParameter() on parameter names with colons

Fixes #8106.

Co-authored-by: Michael Moravec <mail@majkl578.cz>

beberlei committed:
[GH-8106] Move test into existing QueryTest testcase to reduce surface.

orklah committed:
Change list() syntax to array destructuring (short list syntax) (#8204)

orklah committed:
add/fix more types (checked by psalm) (#8199)

* add/fix more psalm types

* remove inexistant SimpleEntityExpression

* Declare template implements for TreeWalkerChainIterator

localheinz committed:
Fix: Typo (#8213)

njutn95 committed:
Update aggregate-fields.rst (#8215)

Minor typo fix

bcremer committed:
Change  preferred cache driver for metadata and query caches (#8223)

See: https://twitter.com/lcobucci/status/1289087725694484481

Commit:
Fix annotation in Expr

greg0ire committed:
Fetch deeper

This is needed to preserve parent information and be able to find the
merge base with the head branch. The assumption here is that the common
ancestor can be found among the 10 parent commits.

greg0ire committed:
Use more accurate terminology (#8236)

I think this was a mistake when writing this documentation, and that the
original author meant to use attribute here, columns do not have
columns, they have attributes.

simPod committed:
Use consistent formatting in command line (#8238)

greg0ire committed:
Stop using the DBAL extension of PHPBench (#8252)

It has been removed, and the default XML storage driver is supposed to
be fine for our purposes.

cziegenberg committed:
Fix for bug #8229 (id column from parent class renamed in child class) (#8234)

This fixes problems with id columns defined in the parent class but renamed in the child class using an attribute override. Before this change always the child column name was used (which was not present in the parent class), now the correct column names are used for the parent table when creating inserts, joins and deletions for it.

Co-authored-by: Crossjoin <crossjoin@users.noreply.github.com>

ThomasLandauer committed:
Update events.rst (#8257)

* Update events.rst

Improved code formatting

* Update docs/en/reference/events.rst

I was wondering myself what the `#` was about ;-)

Co-authored-by: Claudio Zizza <859964+SenseException@users.noreply.github.com>

* Update events.rst

See https://github.com/doctrine/orm/pull/8257#discussion_r482305774

Co-authored-by: Claudio Zizza <859964+SenseException@users.noreply.github.com>

Yohann76 committed:
Improve formatting (#8028)

ipernet committed:
Fix OrderByItem parser to proceed with SimpleArithmetic expression before function (#8277)

There is parser bug for OrderByItem(), where a function is detected
before a SimpleArithmeticExpression while it should be the reverse:
simple arithmetic expressions can start with a function too, and thus
they should be matched first if present.

This fix enables the use of expressions in sorting and with window
functions.

mvorisek committed:
Convert CRLF to LF in test file (#8276)

* Convert CRLF to LF in test file
* fix cs

greg0ire committed:
Use inline literals over escaping (#8279)

Escaping underscores does not work as expected.
See https://www.doctrine-project.org/projects/doctrine-orm/en/latest/cookbook/dql-custom-walkers.html#modify-the-output-walker-to-generate-vendor-specific-sql

orklah committed:
Fix psalm errors and upgrade strictness (#8209)

* Fixes and improvements

* fix param type

orklah committed:
rename parameters to match parents (#8284)

orklah committed:
psalm fixes (#8286)

SenseException committed:
Move website config to default branch (#8287)

Adds the website config to be compatible with the doctrine/doctrine-website#356 changes

greg0ire committed:
Setup automated release workflow (#8301)

beberlei committed:
Change LOG_LEVEL to debug should expose more info from laminas/automatic-releases.

beberlei committed:
Change LOG_LEVEL to debug should expose more info from laminas/automatic-releases.

beberlei committed:
Use SHELL_VERBOSITY=3 instead of LOG_LEVEL=DEBUG

beberlei committed:
Change <phpunit verbose="true" /> to false for better readability of test output.

ThomasLandauer committed:
Update working-with-indexed-associations.rst (#8298)

* Update working-with-indexed-associations.rst

Fixing broken link

* Update docs/en/tutorials/working-with-indexed-associations.rst

Co-authored-by: Claudio Zizza <859964+SenseException@users.noreply.github.com>

Co-authored-by: Claudio Zizza <859964+SenseException@users.noreply.github.com>

l-vo committed:
Fix many typos (#8299)

- Some uppercase letters were used in the middle of sentence
- Some dots were missing
- There was two sentences with wrong or missing words

mvorisek committed:
Fix EOL of text files (#8310)

tomaszgaw committed:
Fix CommitOrderCalculator::addDependency()

It may happen that an entity class A has multiple many-to-one
associations with the same target entity B. These associations may
differ in “nullable” parameter. In such situation
CommitOrderCalculator::addDependency() is called multiple times for the
same pair of $fromHash and $toHash, with different value of $weight
argument. Before the fix the subsequent call to
CommitOrderCalculator::addDependency() for the same $fromHash and
$toHash overrode $weight set at the previous call. It caused
unpredictable commit order of entities which resulted in not null
constraint violation when saving to a database.

Fixed code preserves the highest value of weight argument. If at
subsequent call the weight is lower than previously established weight
the addition of edge to the dependency list is skipped.

greg0ire commented:
Both commits explain what they fix, but not how. Please include the explanation and follow the guidelines at https://www.doctrine-project.org/contribute/index.html#working-on-topic-branches

greg0ire committed:
Upgrade doctrine/coding-standard (#8321)

* Use a classname that exists

Doctrine\ORM\Mapping\TableGenerator does not exist, only
Doctrine\ORM\Id\TableGenerator does.

* Upgrade doctrine/coding-standard

That library has a dependency on another library that requires composer
plugin API v1. Updating both libs allow to use Composer v2.

* Account for doctrine/reflection deprecation

michaelKaefer committed:
Patch 1 (#8325)

* Update outdated doc parts

- The cache implementation moved from `Common` to `doctrine/cache`
- APCu is mor appropiate nowadays I guess
- AbstractQuery::useResultCache() is deprecated since 2.7

* Fix wrong argument

* Fix wrong arguments and remove useless line

beberlei committed:
Start moving travis phpunit runs to Github actions. (#8317)

* Move PHPUnit runs from Travis to Github Actions

This removes all artifacts used for TravisCI testing and replaces them
with the existing infrastructure for Github Actions from DBAL component.

In addition some test changes were needed and triggered larger Coding
Style cleanups in 3 test files.

* Remove composer.lock and improve naming in CI workflow.

beberlei committed:
Ignore composer.lock

greg0ire committed:
Phpunit 8 (#8330)

* Revert to whitelist

coverage requires PHPunit 9, and we don't have that yet.

* Upgrade to PHPUnit 8

This unlocks PCOV usage for coverage

* Upload coverage files to Codecov

rogregoire committed:
Fix SQL alias generation regression for simple inheritance (#8329)

This fixes a regression from 099c5b42e15b4997d50b5b19ffba6fee96c52ff3.
Without the fix, "where part" in SQL is generated with incorrect aliases.
See https://github.com/doctrine/orm/issues/8229#issuecomment-722942180.

RomaVan committed:
Add missing backtick in rst markup (#8335)

SenseException committed:
Update functionality of code examples (#8336)

tomaszgaw committed:
Merge remote-tracking branch 'doctrine/2.7' into fix-commit-order

greg0ire commented:
Closing in favor of #8349
